clients/fides-js/src/lib/consent.tsx

```diff
@@ -0,0 +1,182 @@
+    const body = await response.json();
+    debugLog("Got location response from geolocation API, returning:", body);
+    return body;
+  } catch (e) {
+    debugLog(
+      "Error parsing response body from geolocation API, returning {}. Response:",
+      response
+    );
+    return {};
+  }
+};
+
+/**
+ * Initialize the Fides Consent Banner or Link, with optional extraOptions to override defaults.
+ *
+ * (see the type definition of ConsentBannerOptions for what options are available)
+ */
+export const initFidesConsent = async (
+  defaults: CookieKeyConsent,
+  extraOptions?: ConsentBannerOptions
+): Promise<void> => {
+  debugLog(
+    "Initializing Fides consent banner with consent cookie defaults...",
+    defaults
+  );
+  if (extraOptions !== undefined) {
+    if (typeof extraOptions !== "object") {
+      // eslint-disable-next-line no-console
+      console.error(
+        "Invalid 'extraOptions' arg for Fides.banner(), ignoring",
+        extraOptions
+      );
+    } else {
+      // If the user provides any extra options, override the defaults
+      setBannerOptions({ ...getBannerOptions(), ...extraOptions });
+    }
+  }
+  const options: ConsentBannerOptions = getBannerOptions();
+
+  debugLog("Validating Fides consent banner options...", options);
+  if (!validateBannerOptions(options)) {
+    return Promise.reject(new Error("Invalid banner options"));
+  }
+
+  if (options.isDisabled) {
+    debugLog(
+      "Fides consent banner is disabled, skipping banner initialization!"
+    );
+    return Promise.resolve();
+  }
+
+
+  document.addEventListener("DOMContentLoaded", () => {
+    debugLog("DOM fully loaded and parsed");
+
+    try {
+      debugLog("Adding Fides consent banner CSS & HTML into the DOM...");
+      if (options.isGeolocationEnabled) {
+        getLocation().then(() => {
+          // todo- get applicable notices using location
+        }).catch(() => {
+          // if something goes wrong with location api, we still want to render notices
+        })
+      }
+      const onAcceptAll = () => {
+        setConsentCookieAcceptAll(defaults)
+      }
+      const onRejectAll = () => {
+        setConsentCookieRejectAll(defaults)
+      }
+      render(
+          <ConsentBanner options={options} onAcceptAll={onAcceptAll} onRejectAll={onRejectAll} waitBeforeShow={100}/>
+        ,
+        document.body
+      );
+      const consentLinkEl = document.getElementById("fides-consent-link") as HTMLElement;
+      if (consentLinkEl !== null) {
+        debugLog("Fides consent link el found");
+        consentLinkEl.onclick = () => {
+          debugLog("Navigate to Privacy Center URL:", options.privacyCenterUrl);
+          // todo- depending on notices / experience config, we update onclick of this link to nav to PC or open modal,
+          //  or hide link entirely
+          if (options.privacyCenterUrl) {
+            window.location.assign(options.privacyCenterUrl);
+          }
+        };
+      } else {
```

Comment:
I'll update it to do an `href` swap, although I'm just now thinking through the TODO Catherine left of changing the onClick to open the modal instead. in that case, it might still make sense to use an `onClick`, I'm not sure. ideally it'd be a button in that case though